Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/CollidersMaker.h"
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+#include "RobotImporter/URDF/TypeConversions.h"
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <LmbrCentral/Shape/BoxShapeComponentBus.h>
+#include <LmbrCentral/Shape/CylinderShapeComponentBus.h>
+#include <LmbrCentral/Shape/SphereShapeComponentBus.h>
+#include <SceneAPI/SceneCore/Containers/Scene.h>
+#include <SceneAPI/SceneCore/Containers/SceneManifest.h>
+#include <SceneAPI/SceneCore/Containers/Utilities/Filters.h>
+#include <SceneAPI/SceneCore/DataTypes/Groups/ISceneNodeGroup.h>
+#include <SceneAPI/SceneCore/Events/AssetImportRequest.h>
+#include <SceneAPI/SceneCore/Events/SceneSerializationBus.h>
+#include <SceneAPI/SceneCore/Utilities/SceneGraphSelector.h>
+#include <Source/EditorColliderComponent.h>
+#include <Source/EditorShapeColliderComponent.h>
+
+namespace ROS2
+{
+    CollidersMaker::CollidersMaker(const AZStd::string& modelPath)
+        : m_modelPath(modelPath)
+    {
+    }
+
+    CollidersMaker::~CollidersMaker()
+    {
+    }
+
+    void CollidersMaker::BuildColliders(urdf::LinkSharedPtr link)
+    {
+        for (auto collider : link->collision_array)
+        {
+            BuildCollider(collider);
+        }
+
+        if (link->collision_array.size() == 0)
```

Comment:
Using `.empty()` better documents the intent, that a new entry shuld be added if the collision array is empty.
```suggestion
        if (link->collision_array.empty())
```